recipes/libvpx/all/conanfile.py

```diff
@@ -53,67 +53,99 @@ def config_options(self):
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
 
     def validate(self):
         if self.settings.os == "Windows" and self.options.shared:
             raise ConanInvalidConfiguration("Windows shared builds are not supported")
         if str(self.settings.compiler) not in ["Visual Studio", "msvc", "gcc", "clang", "apple-clang"]:
-            raise ConanInvalidConfiguration("Unsupported compiler {}.".format(self.settings.compiler))
-        if self.settings.os == "Macos" and self.settings.arch == "armv8" and tools.Version(self.version) < "1.10.0":
+            raise ConanInvalidConfiguration(f"Unsupported compiler {self.settings.compiler}")
+        if self.settings.os == "Macos" and self.settings.arch == "armv8" and Version(self.version) < "1.10.0":
             raise ConanInvalidConfiguration("M1 only supported since 1.10, please upgrade")
 
-    @property
-    def _settings_build(self):
-        return getattr(self, "settings_build", self.settings)
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+        # libvpx configure is unusual and requires a workaround, otherwise it will fail with an error.
+        # look for "output_goes_here" below for more in this thread.
+        self.cpp.package.bindirs = []
+        self.cpp.package.includedirs = []
+        self.cpp.package.libdirs = []   # not strictly necessary, but lets do all as a group
```

Comment:
this trick defeats `fix_apple_shared_install_name()` and even the hook checking whether shared libs are relocatables.

Reply:
I might be able to use `tc.update_configure_args({"--bindir":None})` instead (or something like that).
I won't be able to look for another ~12 hours, if you have the time please feel free to fix up and suggest more :)

Reply:
Actually, won't have time to fix, doing the other recipes.  Try my suggestion?  Happy to test your PR locally, tag me in it.